kv/kvcache/cache.go

```diff
@@ -431,6 +442,78 @@ func (c *Coherent) addCode(k, v []byte, r *CoherentRoot, id ViewID) *Element {
 	return it
 }
 
+func (c *Coherent) ValidateCurrentRoot(tx kv.Tx) (*CacheValidationResult, error) {
+	c.lock.Lock()
```

Comment:
likely this func will be very slow - such lock will freeze whole app

Reply:
can't use existing `View(ctx context.Context, tx kv.Tx)` func?

Reply:
no problem, I put the lock in there as the Clone function mentions not to run it concurrently, maybe I could put the lock just around that code rather than the whole func

Reply:
I guess `View()` does lock inside